datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -141,4 +180,23 @@ protected Duration _fromTimestamp(DeserializationContext ctxt, long ts) {
         }
         return Duration.ofMillis(ts);
     }
+
+    protected static class DurationUnitParser {
+        final TemporalUnit unit;
+
+        DurationUnitParser(TemporalUnit unit) {
+            this.unit = unit;
+        }
+
+        Duration parse(long value) {
+            return Duration.of(value, unit);
+        }
+
+        static Optional<DurationUnitParser> from(String unit) {
+            return Stream.of(ChronoUnit.values())
+                    .filter(u -> u.toString().equalsIgnoreCase(unit))
```

Comment:
I think we should not actually allow case-insensitive match, but require upper-case, to exactly match `ChronoUnit`.

Reply:
Now matching is case sensitive.